spec/jobs/hyrax/grant_read_job_spec.rb

```diff
@@ -1,11 +1,22 @@
 RSpec.describe Hyrax::GrantReadJob do
   let(:depositor) { create(:user) }
-  let(:file_set) { build(:file_set) }
+  let(:file_set) { create(:file_set, user: depositor) }
 
-  it 'grants a user read access to a FileSet' do
-    expect(FileSet).to receive(:find).with(file_set.id).and_return(file_set)
-    expect(file_set).to receive(:read_users=).with(array_including(depositor.user_key))
-    expect(file_set).to receive(:save!)
-    described_class.perform_now(file_set.id, depositor.user_key)
+  context "when use_valkyrie is not set" do
+    it 'grants a user read access to a FileSet' do
+      described_class.perform_now(file_set.id, depositor.user_key)
+      file_set.reload
+      expect(file_set.read_users).to include depositor.user_key
+    end
+  end
+
+  [true, false].each do |use_valkyrie|
+    context "when use_valkyrie is #{use_valkyrie}" do
```

Comment:
Do we think it's worth refactoring tests to have a valkyrie native version?

Sooner or later we'll need to do this, so maybe now's the time???

Reply:
Here's my take on the loop for tests as we are in transition from AF to valkyrie.  In a lot of cases, the tests and their outcomes are the same for both approaches.  By using the same tests for both, the AF tests serve as regression tests showing the valkyrie implementation has the same effect.

Also, for these jobs, the tests are super short, but for other classes, the list of tests can be quite long.  Maintaining a set of regression tests for both approaches can be long.

In some cases, the valkyrie test is significantly different and we will have to write a different test.  But when that is not the case, we benefit from the tests remaining the same.

Reply:
Because these job tests are so small, we are going with 2 tests... 1) pure AF, 2) pure valkyrie